tiledb/sm/serialization/group.cc

```diff
@@ -70,7 +70,7 @@ Status group_metadata_to_capnp(
   auto config_builder = group_metadata_builder->initConfig();
   RETURN_NOT_OK(config_to_capnp(group->config(), &config_builder));
 
-  const Metadata* metadata = group->metadata();
+  const Metadata* metadata = const_cast<Group*>(group)->metadata();
```

Comment:
Can't we remove const on the input of group_metadata_to_capnp, tracking the calling chain up, it looks like at some point group is not const anymore.

Reply:
Should we swap to use `Status Group::metadata(Metadata** metadata)` here and below?